nvm.sh

```diff
@@ -2405,8 +2406,17 @@ nvm() {
       nvm_err "uname -a: '$(uname -a | awk '{$2=""; print}' | xargs)'"
       if [ "$(nvm_get_os)" = "darwin" ] && nvm_has sw_vers; then
         nvm_err "OS version: $(sw_vers | command awk '{print $2}' | command xargs)"
-      elif [ -r "/etc/issue" ]; then
-        nvm_err "OS version: $(command head -n 1 /etc/issue | command sed 's/\\.//g')"
+      else
+        if [ -r "/etc/issue" ]; then
+          nvm_err "OS version: $(command head -n 1 /etc/issue | command sed 's/\\.//g')"
+        fi
+        if [ -r "/etc/os-release" ] && [ "${OS_VERSION}" = "" ] ; then
+          # shellcheck disable=SC1091
+          OS_VERSION="$(. /etc/os-release && echo "${NAME}" "${VERSION}")"
```

Comment:
it seems risky to be sourcing a random file; is this necessary?

Reply:
Is it random? I don't think so. It's a built-in of most of the modern Linux distros, and it has clearly definition: https://www.freedesktop.org/software/systemd/man/os-release.html